tests/resources/helper_functions.py

```diff
@@ -186,3 +194,74 @@ def verify_supply(mint_amount=to_wei(50)):
                 break
         except (ValueError, Exception):
             pass
+
+
+def deploy_erc721_erc20(
+    web3: Web3, config: Config, erc721_publisher: Wallet, erc20_minter: Wallet
+):
+    """Helper function to deploy an ERC721Token using erc721_publisher Wallet
+    and an ERC20Token data token with the newly ERC721Token using erc20_minter Wallet.
+
+    :rtype: (ERC721Token, ERC20Token)
+    """
+
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+    tx = erc721_factory.deploy_erc721_contract(
+        "NFT",
+        "NFTSYMBOL",
+        1,
+        ZERO_ADDRESS,
+        "https://oceanprotocol.com/nft/",
+        erc721_publisher,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_NFT_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    token_address = registered_event[0].args.newTokenAddress
+    erc721_token = ERC721Token(web3, token_address)
+
+    erc_create_data = ErcCreateData(
+        1,
+        ["ERC20DT1", "ERC20DT1Symbol"],
+        [
+            erc20_minter.address,
+            erc721_publisher.address,
+            erc721_publisher.address,
+            ZERO_ADDRESS,
+        ],
+        [web3.toWei("0.5", "ether"), 0],
+        [b""],
+    )
+    tx_result = erc721_token.create_erc20(erc_create_data, erc721_publisher)
+    tx_receipt2 = web3.eth.wait_for_transaction_receipt(tx_result)
+
+    registered_event2 = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_TOKEN_CREATED,
+        tx_receipt2.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    erc20_address = registered_event2[0].args.newTokenAddress
+
+    erc20_token = ERC20Token(web3, erc20_address)
+
+    return erc721_token, erc20_token
+
+
+def get_non_existent_nft_template(
+    erc721_factory: ERC721FactoryContract, check_first=20
```

Comment:
I ended up creating this helper to find non existent templates in response to this comment: https://github.com/oceanprotocol/ocean.py/pull/546/files/b8ab450edeccbd6527001856fe6be0f2769806be#r740784915